activerecord/lib/active_record/connection_adapters/sqlite3/column.rb

```diff
@@ -20,6 +21,18 @@ def auto_incremented_by_db?
           auto_increment? || rowid
         end
 
+        def virtual?
+          @generated_type.present?
```

Comment:
```suggestion
          @generated_type
```

`.present?` isn't a "to_boolean" cast, it has fairly peculiar rules, and is rarely what you want.

Here we can just return `@generated_type`, or if you are adamant  about returning a boolean, you can use `!!@generated_type` or `!@generated_type.nil?`